test/unit/Cover/buyCover.js

```diff
@@ -572,7 +572,60 @@ describe('buyCover', function () {
       {
         value: expectedPremium,
       },
-    )).to.be.revertedWith('Cover: amount = 0');
+    )).to.be.revertedWith('Cover: coverAmountInAsset = 0');
 
   });
+
+  it('should revert when the allocated cover amount is less than the expected cover amount', async function () {
+    const { cover } = this;
+
+    const {
+      governanceContracts: [gv1],
+      members: [member1],
+      members: [coverBuyer1, stakingPoolManager],
+    } = this.accounts;
+
+    const productId = 0;
+    const payoutAsset = 0; // ETH
+    const period = 3600 * 24 * 364; // 30 days
+
+    const amount = parseEther('1000');
+
+    const targetPriceRatio = '260';
+    const priceDenominator = '10000';
+    const activeCover = parseEther('8000');
+    const capacity = parseEther('10000');
+
+    const capacityFactor = '10000';
+
+    await cover.connect(gv1).updateUintParameters([0], [capacityFactor]);
+
+    await createStakingPool(
+      cover, productId, capacity, targetPriceRatio, activeCover, stakingPoolManager, stakingPoolManager, targetPriceRatio,
+    );
+
+    const expectedPremium = amount.mul(targetPriceRatio).div(priceDenominator);
+
+    const tooLargeExpectedAmount = amount.add(10);
+
+    await expect(cover.connect(member1).buyCover(
+      {
+        owner: coverBuyer1.address,
+        productId,
+        payoutAsset,
+        amount: tooLargeExpectedAmount,
+        period,
+        maxPremiumInAsset: expectedPremium,
+        paymentAsset: payoutAsset,
+        payWitNXM: false,
+        commissionRatio: parseEther('0'),
+        commissionDestination: ZERO_ADDRESS,
+        ipfsData: ''
+      },
+      [{ poolId: '0', coverAmountInAsset: amount.toString() }],
+      {
+        value: expectedPremium,
+      },
+    )).to.be.revertedWith('Cover Insufficient cover amount');
```

Comment:
:hammer: The revert reason wording suggests that the user didn't buy enough cover when in reality we ran out of capacity on some of the provided pools. Let's change the reason.

I'd suggest something like `Cover: The selected pools ran out of capacity`